internal/util/helpers_test.go

```diff
@@ -13,21 +13,765 @@
+				t.Errorf("GetYamlFromProto() error = %v, expectedError %v", err, tt.expectedError)
+				return
+			}
+			if yamlResult != tt.expectedYaml {
+				t.Errorf("GetYamlFromProto() = %v, expected %v", yamlResult, tt.expectedYaml)
+			}
+		})
+	}
+}
+
+// TestOpenFileArg tests the OpenFileArg function
+func TestOpenFileArg(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name         string
+		filePath     string
+		dashOpen     io.Reader
+		expectedDesc string
+		expectError  bool
+	}{
+		{
+			name:         "Dash as file path",
+			filePath:     "-",
+			dashOpen:     strings.NewReader("dash input"),
+			expectedDesc: "dash input",
+			expectError:  false,
+		},
+		{
+			name:         "Valid file path",
+			filePath:     "testfile.txt",
+			dashOpen:     nil,
+			expectedDesc: "test content",
+			expectError:  false,
+		},
+		{
+			name:         "Invalid file path",
+			filePath:     "nonexistent.txt",
+			dashOpen:     nil,
+			expectedDesc: "",
+			expectError:  true,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			// Create a temporary file for testing
+			if tc.filePath == "testfile.txt" {
+				err := os.WriteFile(tc.filePath, []byte(tc.expectedDesc), 0600)
+				assert.NoError(t, err)
+				defer os.Remove(tc.filePath)
+			}
+
+			desc, closer, err := OpenFileArg(tc.filePath, tc.dashOpen)
+			if closer != nil {
+				defer closer()
+			}
+
+			if tc.expectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				if tc.filePath == "-" || tc.filePath != "-" {
+					buf := new(strings.Builder)
+					_, err := io.Copy(buf, desc)
+					assert.NoError(t, err)
+					assert.Equal(t, tc.expectedDesc, buf.String())
+				}
+			}
+		})
+	}
+}
+
+// TestExpandFileArgs tests the ExpandFileArgs function.
+func TestExpandFileArgs(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name     string
+		files    []string
+		expected []string
+		wantErr  bool
+	}{
+		{
+			name:     "Single file",
+			files:    []string{"testfile.txt"},
+			expected: []string{"testfile.txt"},
+			wantErr:  false,
+		},
+		{
+			name:     "Single directory",
+			files:    []string{"testdir"},
+			expected: []string{"testdir/file1.txt", "testdir/file2.txt"},
+			wantErr:  false,
+		},
+		{
+			name:     "File and directory",
+			files:    []string{"testfile.txt", "testdir"},
+			expected: []string{"testfile.txt", "testdir/file1.txt", "testdir/file2.txt"},
+			wantErr:  false,
+		},
+		{
+			name:     "File with '-'",
+			files:    []string{"-"},
+			expected: []string{"-"},
+			wantErr:  false,
+		},
+		{
+			name:     "Non-existent file",
+			files:    []string{"nonexistent.txt"},
+			expected: nil,
+			wantErr:  true,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			// Create a unique directory for each test
+			testDir := fmt.Sprintf("testdir_%s", sanitizeTestName(tt.name))
```

Comment:
You could probably leverage golang's utilities instead: https://pkg.go.dev/testing#B.TempDir

This will create a temporary directory for tests and even clean up afterwards.

wdyt?

Reply:
thank you. I agree. that is a clean approach.